src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -189,25 +220,34 @@ class NwbRecordingExtractor(BaseRecording):
 
     def __init__(
         self,
-        file_path: str | Path,
-        electrical_series_name: str = None,
+        file_path: str | Path | None = None,  # provide either this or file
+        electrical_series_name: str | None = None,
         load_time_vector: bool = False,
         samples_for_rate_estimation: int = 100000,
-        stream_mode: Optional[Literal["fsspec", "ros3"]] = None,
+        stream_mode: Optional[Literal["fsspec", "ros3", "remfile"]] = None,
```

Comment:
pylance linter complains when Optional is left out, so I think the correct thing to do is to leave it in.

Reply:
I think the linters are behind the times, but fair enough. :)

Reply:
Really? I thought that was the meaning of Optional. When something can be None.

Reply:
I also thought that `Optional` means that it defaults to `None`. Not really sure about it though now

Reply:
The documentation says that this is the typing for when the value can be None (independentl on whether the argument has a default or not):

https://docs.python.org/3/library/typing.html#typing.Optional

So I think what @magland is doing the correct typing here.  It could also be `Literal["fsspec", "ros3", "remfile", None]`

Reply:
Not to rehash, but I think the issue is also that typing has a 3.8 variety (Optional, Literal etc) and a 3.10+ variety with | (so a | b | None), but SI has a mix of 3.8 style and 3.10 style and I don't think mixing styles would play the best with linters, not to mention that Python keeps allowing more and more type hint styles. So I can't imagine the linters can keep up. 

I personally think that the `|` notation is easier to read as an end user, but more important is just having the hints somewhere to help guide the user what to use.